src/event/player/PlayerCreationEvent.php

```diff
@@ -118,18 +117,24 @@ public function setPlayerClass(string $class) : void{
 	}
 
 	/**
-	 * The typical use cases for this WaitGroup include:
-	 *  - Carrying out preparatory tasks such as configuring default preferences, setting
-	 * player-specific settings, or initializing in-game assets.
-	 *  - Loading initial player data from external sources or databases before the login sequence begins.
-	 *  - Providing a tailored welcome experience, which might involve sending messages,
-	 * granting starter items, or awarding initial rewards.
+	 * Retrieves all promises that have been added to the event.
+	 * These promises must be completed before the player login sequence is initiated.
 	 *
-	 * Note:
-	 *  After each task is completed, it's essential to call the `done` method of the WaitGroup
-	 *  to signal its completion and allow the WaitGroup to determine when to proceed.
+	 * @return Promise[]
+	 * @phpstan-return Promise<mixed>[]
 	 */
-	public function getWaitGroup() : WaitGroup{
-		return $this->waitGroup;
+	public function getPromises() : array{
```

Comment:
Doubtful it makes sense to expose this

Reply:
Funny you should mention that, I thought the same thing and it's fixed in 64a164d7aa300429b1d6a927626647d291bfc342